### ISSUE TITLE
Cmake error on fresh build

### ISSUE THREAD

**Issue by nvnmo** [google/benchmark#810]
Cmake error while configuring a fresh build following the [installation instructions](https://github.com/google/benchmark/blob/master/README.md#installation).

```
-- Performing Test HAVE_POSIX_REGEX -- success
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK -- success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
CMake Error at src/CMakeLists.txt:37 (string):
  string no output variable specified


-- Looking for Google Test sources

```
It is similar to #124 and goes away when I set `CMAKE_BUILD_TYPE`.

The variable is referenced without checking [here](https://github.com/google/benchmark/blob/823d24630db0f340914bb011f1ba31657348e510/src/CMakeLists.txt#L37).

Seems like CMAKE_BUILD_TYPE doesn't have any default value as mentioned [here](https://github.com/google/benchmark/blob/master/README.md#debug-vs-release). 

Platform: linux
Compiler: GCC 7.8
Cmake: 3.14.2

**Comment by LebedevRI** [comment]
Right, let's add a default.

**Comment by nvnmo** [comment]
Fixed by PR #811.
